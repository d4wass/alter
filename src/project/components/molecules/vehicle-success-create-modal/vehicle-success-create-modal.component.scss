@use '../../../../base';

.wrapper {
  position: fixed;
  z-index: 10;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(51, 51, 51, 0.6);
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
}

.wrapper-modal {
  border-radius: base.$main-border-rad;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 30px;
  background-color: #fff;
  width: 35%;
}

.wrapper-modal > * {
  width: 100%;
}

.title {
  text-align: center;
}

.buttons,
.inputs {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.standard-btn {
  cursor: pointer;
  width: 100%;
  padding: 20px 0;
  border-radius: base.$main-border-rad;
  border: none;
  background-color: #000;
  color: #fff;
  font-weight: 700;
  margin: 30px 0;
}

.cancel-btn {
  text-decoration: underline;
  font-weight: 700;
  text-transform: capitalize;
  border: none;
  width: 100%;
  padding: 20px 0;
  background-color: transparent;
  cursor: pointer;
}

.divider {
  display: flex;
  align-items: center;
  margin: 30px 0;
  p {
    color: base.$light-grey;
    font-size: 25px;
    font-weight: 700;
    margin: 0 20px;
  }
  span {
    content: '';
    height: 2px;
    width: 100%;
    background-color: base.$light-grey;
  }
}

.social-btn {
  padding: 20px 20px;
  margin-bottom: 15px;
  border-radius: base.$main-border-rad;
  border: solid 2px base.$light-grey;
  background-color: transparent;
  font-weight: 700;
  text-align: start;
  display: flex;
  align-items: center;
  cursor: pointer;

  img {
    height: 25px;
  }

  p {
    margin: 0;
    margin-left: 10px;
  }
}

.change-btn {
  display: flex;
  justify-content: center;
  align-items: center;
  p {
    font-weight: 700;
    margin-right: 20px;
  }
  button {
    cursor: pointer;
    padding: 5px 10px;
    background-color: transparent;
    border: 1px #000 solid;
    font-size: 13px;
    border-radius: base.$main-border-rad;
  }
}

input {
  width: 100%;
  padding: 15px 20px;
  border-radius: base.$main-border-rad;
  border: solid 2px base.$light-grey;
  margin-bottom: 20px;

  &::placeholder {
    color: base.$light-grey;
  }
}

.personal {
  padding: 15px 20px;
  border-radius: base.$main-border-rad;
  border: solid 2px base.$light-grey;
  width: 100%;

  &::placeholder {
    color: base.$light-grey;
  }
}

.invalid {
  border: solid 2px red;
}
