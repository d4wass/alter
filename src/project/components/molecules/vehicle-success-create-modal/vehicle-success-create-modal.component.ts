import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'app-vehicle-success-create-modal',
  template: `
    <ng-container class="wrapper" *ngIf="isCreated">
      <div class="wrapper-modal">
        <h2>Your Vehicle was successfully created</h2>
        <button (click)="handleCloseBtn()">close</button>
      </div>
    </ng-container>
  `,
  styleUrls: ['./vehicle-success-create-modal.component.scss']
})
export class VehicleSuccessCreateModalComponent {
  @Input() isCreated: boolean = false;
  @Output() handleClick: EventEmitter<any> = new EventEmitter<any>();

  constructor() {}

  handleCloseBtn(): void {
    this.handleClick.emit();
  }
}
