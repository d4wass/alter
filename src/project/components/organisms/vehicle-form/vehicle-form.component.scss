.wrapper {
  padding: 0 6.3em;
  display: flex;
  flex-direction: column;
}
