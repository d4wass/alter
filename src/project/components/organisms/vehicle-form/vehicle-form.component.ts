import { ChangeDetectionStrategy, Component } from '@angular/core';
import { Validators } from '@angular/forms';
import { Router } from '@angular/router';
import { ControlsOf, FormControl, FormGroup } from '@ngneat/reactive-forms';
import { Store } from '@ngrx/store';
import { filter, map, Observable, switchMap, take, withLatestFrom } from 'rxjs';
import { UserActions } from 'src/+state/user/user.actions';
import {
  IVehicleBasicData,
  IVehicleFeaturesData,
  IVehicleSpecificationData
} from 'src/project/model/vehicle-form-models/vehicle-forms.model';
import { UserFacade } from '../../../../+state/facade/user.facade';

@Component({
  selector: 'app-vehicle-form',
  template: `
    <div class="wrapper">
      <form [formGroup]="vehicleForm">
        <app-vehicle-form-main-data
          [formGroupCtrl]="vehicleForm.get('vehicleMainInfo')"
        ></app-vehicle-form-main-data>
        <app-vehicle-form-spec-data
          [formGroupCtrl]="vehicleForm.get('vehicleSpecInfo')"
        ></app-vehicle-form-spec-data>
        <app-vehicle-form-features-data
          [formGroupCtrl]="vehicleForm.get('vehicleFeaturesInfo')"
        ></app-vehicle-form-features-data>
        <button (click)="handleSubmitVehicle()">add vehicle</button>
      </form>
      <app-vehicle-success-create-modal
        [isCreated]="isCreated | async"
        (handleClick)="resetFormAfterCreation()"
      ></app-vehicle-success-create-modal>
    </div>
  `,
  styleUrls: ['./vehicle-form.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class VehicleFormComponent {
  isCreated: Observable<boolean> = this.userFacade.isVehicleCreated();
  vehicleForm: FormGroup<ControlsOf<any>> = new FormGroup<ControlsOf<any>>({
    vehicleMainInfo: new FormGroup<ControlsOf<IVehicleBasicData>>({
      brand: new FormControl('', [Validators.required, Validators.minLength(3)]),
      model: new FormControl('', [Validators.required, Validators.minLength(3)]),
      place: new FormControl('', [Validators.required, Validators.minLength(3)]),
      price: new FormControl(null, [Validators.required, Validators.minLength(3)]),
      description: new FormControl('')
    }),
    vehicleSpecInfo: new FormGroup<ControlsOf<IVehicleSpecificationData>>({
      fuelConsumption: new FormGroup<{ units: FormControl<string>; quantity: any }>({
        units: new FormControl('', Validators.required),
        quantity: new FormControl('', Validators.required)
      }),
      fuelType: new FormControl('', Validators.required),
      doors: new FormControl(0, Validators.required),
      seats: new FormControl(0, Validators.required)
    }),
    vehicleFeaturesInfo: new FormGroup<ControlsOf<IVehicleFeaturesData>>({
      engine: new FormGroup<{ capacity: FormControl<string>; power: FormControl<number | null> }>({
        capacity: new FormControl('', [Validators.required]),
        power: new FormControl(null, [Validators.required])
      }),
      gearbox: new FormControl<string>('manual', Validators.required),
      drive: new FormControl<string>('rear', Validators.required),
      //rid off that any types
      equipment: new FormGroup<ControlsOf<any>>({
        appleCarPlay: new FormControl(false),
        androidAuto: new FormControl(false),
        bluetooth: new FormControl(false),
        usb: new FormControl(false),
        headUpDisplay: new FormControl(false),
        navigation: new FormControl(false),
        airConditioning: new FormControl(false),
        keyLess: new FormControl(false),
        isofix: new FormControl(false),
        cruiseControl: new FormGroup<{
          standard: FormControl<boolean>;
          active: FormControl<boolean>;
          adaptive: FormControl<boolean>;
          autonomic: FormControl<boolean>;
        }>({
          standard: new FormControl(false),
          active: new FormControl(false),
          adaptive: new FormControl(false),
          autonomic: new FormControl(false)
        }),
        lights: new FormGroup<{
          led: FormControl<boolean>;
          xenon: FormControl<boolean>;
          biXenon: FormControl<boolean>;
          laser: FormControl<boolean>;
        }>({
          led: new FormControl(false),
          xenon: new FormControl(false),
          biXenon: new FormControl(false),
          laser: new FormControl(false)
        }),
        parkingAssist: new FormGroup<{
          camera: FormControl<boolean>;
          camera360: FormControl<boolean>;
          autonomic: FormControl<boolean>;
        }>({
          camera: new FormControl(false),
          camera360: new FormControl(false),
          autonomic: new FormControl(false)
        })
      })
    })
  });

  constructor(
    private readonly store: Store,
    private readonly userFacade: UserFacade,
    private readonly router: Router
  ) {}

  handleSubmitVehicle(): void {
    this.vehicleForm.value$
      .pipe(take(1))
      .subscribe((vehicle) => this.store.dispatch(UserActions.addUserVehicle({ vehicle })));
  }

  resetFormAfterCreation(): void {
    this.userFacade
      .isVehicleCreated()
      .pipe(
        filter((isCreated) => isCreated),
        take(1)
      )
      .subscribe(() => {
        this.store.dispatch(UserActions.vehicleResetForm());
        this.router.navigate([`profile`]);
        this.vehicleForm.reset();
      });
  }
}
