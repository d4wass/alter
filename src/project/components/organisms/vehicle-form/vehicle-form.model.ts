import { FormControl, FormGroup } from '@ngneat/reactive-forms';

export interface VehicleForm {
  vehicleMainInfo: FormGroup<VehicleMainInformation>;
  vehicleSpecInfo: FormGroup<VehicleSpecificationInformation>;
  vehicleFeaturesInfo: FormGroup<any>;
}

export interface VehicleMainInformation {
  brand: FormControl<string>;
  model: FormControl<string>;
  place: FormControl<string>;
  price: FormControl<number | null>;
  description: FormControl<string>;
}

export interface VehicleSpecificationInformation {
  fuelConsumption: FormGroup<FuelConsumption>;
  fuelType: FormControl<string>;
  doors: FormControl<number | null>;
  seats: FormControl<number | null>;
}

export interface VehicleFeaturesInformation {
  engine: FormGroup<Engine>;
  gearbox: FormControl<string>;
  drive: FormControl<string>;
  equipment: FormControl<string[]>;
}

type FuelConsumption = {
  units: FormControl<string>;
  quantity: FormControl<string>;
};

type Engine = {
  capacity: FormControl<string>;
  power: FormControl<number | null>;
};
