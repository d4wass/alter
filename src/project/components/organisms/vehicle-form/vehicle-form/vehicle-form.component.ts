import { ChangeDetectionStrategy, Component } from '@angular/core';
import { Validators } from '@angular/forms';
import { Router } from '@angular/router';
import { FormControl, FormGroup } from '@ngneat/reactive-forms';
import { Store } from '@ngrx/store';
import { filter, Observable, take, tap } from 'rxjs';
import { UserActions } from '../../../../../+state/user/user.actions';
import { UserFacade } from '../../../../../+state/facade/user/user.facade';
import {
  VehicleFeaturesInformation,
  VehicleForm,
  VehicleMainInformation,
  VehicleSpecificationInformation
} from '../vehicle-form.model';

@Component({
  selector: 'app-vehicle-form',
  template: `
    <div class="wrapper">
      <form [formGroup]="form">
        <app-vehicle-form-main-data
          [formGroupCtrl]="form.get('vehicleMainInfo')"
        ></app-vehicle-form-main-data>
        <app-vehicle-form-spec-data
          [formGroupCtrl]="form.get('vehicleSpecInfo')"
        ></app-vehicle-form-spec-data>
        <app-vehicle-form-features-data
          [formGroupCtrl]="form.get('vehicleFeaturesInfo')"
        ></app-vehicle-form-features-data>
        <button (click)="handleSubmitVehicle()" data-test="submit-btn">add vehicle</button>
      </form>
      <app-vehicle-success-create-modal
        [isCreated]="isCreated$ | async"
        (handleClick)="resetFormAfterCreation()"
      ></app-vehicle-success-create-modal>
    </div>
  `,
  styleUrls: ['./vehicle-form.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class VehicleFormComponent {
  isCreated$: Observable<boolean> = this.userFacade.isVehicleCreated();

  brandCtrl = new FormControl('', [Validators.required, Validators.minLength(3)]);
  modelCtrl = new FormControl('', [Validators.required, Validators.minLength(3)]);
  placeCtrl = new FormControl('', [Validators.required, Validators.minLength(3)]);
  priceCtrl = new FormControl(null, [Validators.required, Validators.minLength(3)]);
  descriptionCtrl = new FormControl('');

  unitsCtrl = new FormControl('', Validators.required);
  quantityCtrl = new FormControl('', Validators.required);
  fuelTypeCtrl = new FormControl('', Validators.required);
  doorsCtrl = new FormControl(0, Validators.required);
  seatsCtrl = new FormControl(0, Validators.required);

  fuelConsumption = new FormGroup({
    units: this.unitsCtrl,
    quantity: this.quantityCtrl
  });

  capacityCtrl = new FormControl('', [Validators.required]);
  powerCtrl = new FormControl(null, [Validators.required]);
  gearboxCtrl = new FormControl<string>('', Validators.required);
  driveCtrl = new FormControl<string>('', Validators.required);
  equipmentCtrl = new FormControl<string[]>([]);

  engine = new FormGroup({
    capacity: this.capacityCtrl,
    power: this.powerCtrl
  });

  vehicleMainInformation = new FormGroup<VehicleMainInformation>({
    brand: this.brandCtrl,
    model: this.modelCtrl,
    place: this.placeCtrl,
    price: this.priceCtrl,
    description: this.descriptionCtrl
  });
  vehicleSpecificationInformation = new FormGroup<VehicleSpecificationInformation>({
    fuelConsumption: this.fuelConsumption,
    fuelType: this.fuelTypeCtrl,
    doors: this.doorsCtrl,
    seats: this.seatsCtrl
  });
  vehicleFeaturesInformation = new FormGroup<VehicleFeaturesInformation>({
    engine: this.engine,
    gearbox: this.gearboxCtrl,
    drive: this.driveCtrl,
    equipment: this.equipmentCtrl
  });

  form = new FormGroup<VehicleForm>({
    vehicleMainInfo: this.vehicleMainInformation,
    vehicleSpecInfo: this.vehicleSpecificationInformation,
    vehicleFeaturesInfo: this.vehicleFeaturesInformation
  });

  constructor(
    private readonly store: Store,
    private readonly userFacade: UserFacade,
    private readonly router: Router
  ) {}

  handleSubmitVehicle(): void {
    this.form.value$
      .pipe(take(1))
      .subscribe((vehicle) => this.store.dispatch(UserActions.addUserVehicle({ vehicle })));
  }

  resetFormAfterCreation(): void {
    this.userFacade
      .isVehicleCreated()
      .pipe(
        tap((x) => console.log(x)),
        filter((isCreated) => isCreated),
        take(1)
      )
      .subscribe(() => {
        this.store.dispatch(UserActions.vehicleResetForm());
        this.router.navigate([`profile`]);
        this.form.reset();
      });
  }
}
