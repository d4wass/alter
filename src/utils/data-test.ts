export function dataTest(selector: string) {
  return `[data-test="${selector}"]`;
}
