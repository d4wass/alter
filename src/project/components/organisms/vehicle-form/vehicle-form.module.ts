import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VehicleFormComponent } from './vehicle-form/vehicle-form.component';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatInputModule } from '@angular/material/input';
import { MatSelectModule } from '@angular/material/select';
import { MatRadioModule } from '@angular/material/radio';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { ReactiveFormsModule } from '@angular/forms';
import { VehicleFormMainDataModule } from '../../molecules/vehicle-form-main-data/vehicle-form-main-data.module';
import { VehicleFormSpecDataModule } from '../../molecules/vehicle-form-spec-data/vehicle-form-spec-data.module';
import { VehicleFormFeaturesDataModule } from '../../molecules/vehicle-form-features-data/vehicle-form-features-data.module';
import { VehicleSuccessCreateModalComponent } from '../../molecules/vehicle-success-create-modal/vehicle-success-create-modal.component';

@NgModule({
  declarations: [VehicleFormComponent, VehicleSuccessCreateModalComponent],
  imports: [
    CommonModule,
    MatFormFieldModule,
    MatInputModule,
    MatSelectModule,
    MatRadioModule,
    MatCheckboxModule,
    VehicleFormMainDataModule,
    VehicleFormSpecDataModule,
    VehicleFormFeaturesDataModule,
    ReactiveFormsModule
  ],
  exports: [VehicleFormComponent]
})
export class VehicleFormModule {}
