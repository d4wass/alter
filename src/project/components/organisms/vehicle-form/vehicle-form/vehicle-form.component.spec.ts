import { Router } from '@angular/router';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { Store } from '@ngrx/store';
import { MockComponent, MockProvider } from 'ng-mocks';
import { of } from 'rxjs';
import { UserActions } from '../../../../../+state/user/user.actions';
import { UserFacade } from '../../../../../+state/facade/user/user.facade';
import { dataTest } from '../../../../../utils/data-test';
import { VehicleFormComponent } from './vehicle-form.component';
import { mockVehicleFormData } from '../../../../../utils/mocks/mockVehicleFormData';
import { VehicleSuccessCreateModalComponent } from '../../../../components/molecules/vehicle-success-create-modal/vehicle-success-create-modal.component';
import { MOCK_USER_FACADE, UserFacadeMock } from '../../../../../utils/mocks/user.facade.mock';

describe('VehicleFormComponent', () => {
  let component: VehicleFormComponent;
  let vehicleSuccessCreateComponent: VehicleSuccessCreateModalComponent | null;
  let spectator: Spectator<VehicleFormComponent>;
  let store: Store;
  let userFacade: UserFacadeMock;
  let router: Router;

  const createComponent = createComponentFactory({
    component: VehicleFormComponent,
    declarations: [MockComponent(VehicleSuccessCreateModalComponent)],
    providers: [
      MockProvider(Router),
      { provide: Store, useValue: { dispatch: jest.fn() } },
      {
        provide: UserFacade,
        useValue: {
          isVehicleCreated: () => of(false)
        }
      },
      MOCK_USER_FACADE
    ],
    shallow: true
  });

  const selectors = {
    submitBtn: () => spectator.query<HTMLButtonElement>(dataTest('submit-btn'))
  };

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
    store = spectator.inject(Store);
    userFacade = spectator.inject(UserFacadeMock);
    router = spectator.inject(Router);
    vehicleSuccessCreateComponent = spectator.query(VehicleSuccessCreateModalComponent);
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match to snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  describe('handleSubmitVehicle()', () => {
    it('should dispatch action on fn call', () => {
      const vehicle = mockVehicleFormData();
      component.handleSubmitVehicle();
      expect(store.dispatch).toHaveBeenCalledWith(UserActions.addUserVehicle({ vehicle }));
    });
    it('should call fn on button click', () => {
      const spy = jest.spyOn(component, 'handleSubmitVehicle');
      spectator.dispatchMouseEvent(selectors.submitBtn()!, 'click');

      expect(spy).toBeCalled();
    });
  });

  describe('resetFormAfterCreation()', () => {
    beforeEach(() => jest.resetAllMocks());

    it('should reset form on (handleClick) emitter', () => {
      const spy = jest.spyOn(component.form, 'reset');
      userFacade.vehicleCreated$.next(true);
      vehicleSuccessCreateComponent?.handleClick.emit();

      expect(spy).toBeCalled();
    });
    it('should not reset form when vehicle was not successfuly created', () => {
      const spy = jest.spyOn(component.form, 'reset');
      vehicleSuccessCreateComponent?.handleClick.emit();

      expect(spy).not.toBeCalled();
    });
    it('should dispatch action after vehicle creation', () => {
      userFacade.vehicleCreated$.next(true);
      vehicleSuccessCreateComponent?.handleClick.emit();

      expect(store.dispatch).toBeCalledWith(UserActions.vehicleResetForm());
    });
    it('should not dispatch action when vehicle was not successfuly created', () => {
      vehicleSuccessCreateComponent?.handleClick.emit();

      expect(store.dispatch).not.toBeCalledWith(UserActions.vehicleResetForm());
    });

    it.each([true, false])(
      'should call router navigate depend on successfully vehicle creation',
      (vehicleCreated: boolean) => {
        const spy = jest.spyOn(router, 'navigate');
        userFacade.vehicleCreated$.next(vehicleCreated);
        vehicleSuccessCreateComponent?.handleClick.emit();

        if (vehicleCreated) {
          expect(spy).toHaveBeenCalledWith(['profile']);
        } else {
          expect(spy).not.toHaveBeenCalled();
        }
      }
    );
  });
});
