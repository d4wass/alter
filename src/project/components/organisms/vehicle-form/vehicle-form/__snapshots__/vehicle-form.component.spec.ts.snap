// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleFormComponent should match to snapshot 1`] = `
<app-vehicle-form
  brandCtrl={[Function FormControl]}
  capacityCtrl={[Function FormControl]}
  descriptionCtrl={[Function FormControl]}
  doorsCtrl={[Function FormControl]}
  driveCtrl={[Function FormControl]}
  engine={[Function FormGroup]}
  equipmentCtrl={[Function FormControl]}
  form={[Function FormGroup]}
  fuelConsumption={[Function FormGroup]}
  fuelTypeCtrl={[Function FormControl]}
  gearboxCtrl={[Function FormControl]}
  isCreated$={[Function BehaviorSubject]}
  modelCtrl={[Function FormControl]}
  placeCtrl={[Function FormControl]}
  powerCtrl={[Function FormControl]}
  priceCtrl={[Function FormControl]}
  quantityCtrl={[Function FormControl]}
  router={[Function Router]}
  seatsCtrl={[Function FormControl]}
  store={[Function Object]}
  unitsCtrl={[Function FormControl]}
  userFacade={[Function UserFacadeMock]}
  vehicleFeaturesInformation={[Function FormGroup]}
  vehicleMainInformation={[Function FormGroup]}
  vehicleSpecificationInformation={[Function FormGroup]}
>
  <div
    class="wrapper"
  >
    <form>
      <app-vehicle-form-main-data />
      <app-vehicle-form-spec-data />
      <app-vehicle-form-features-data />
      <button
        data-test="submit-btn"
      >
        add vehicle
      </button>
    </form>
    <app-vehicle-success-create-modal />
  </div>
</app-vehicle-form>
`;
